sdk/examples/wallet/spammer.rs

```diff
@@ -0,0 +1,198 @@
+    },
+    wallet::{account::types::AccountAddress, Account, ClientOptions, Result, SendAmountParams, Wallet},
+};
+use tokio::{task::JoinSet, time::sleep};
+
+// The account alias used in this example.
+const ACCOUNT_ALIAS: &str = "spammer";
+// The number of spamming rounds.
+const NUM_ROUNDS: usize = 1000;
+// The amount to send in each transaction.
+const SEND_AMOUNT: u64 = 1_000_000;
+// The number of concurrent transactions.
+const NUM_CONCURRENT_TXS: usize = 32;
+
+#[tokio::main(flavor = "multi_thread")]
+async fn main() -> Result<()> {
+    // This example uses secrets in environment variables for simplicity which should not be done in production.
+    dotenvy::dotenv().ok();
+
+    println!(
+        "{} cores detected. Running spammer with {NUM_CONCURRENT_TXS} concurrent transactions...",
+        num_cpus::get(),
+    );
+
+    // Restore wallet from a mnemonic phrase.
+    let client_options = ClientOptions::new().with_node(&std::env::var("NODE_URL").unwrap())?;
+    let secret_manager =
+        MnemonicSecretManager::try_from_mnemonic(&std::env::var("NON_SECURE_USE_OF_DEVELOPMENT_MNEMONIC_1").unwrap())?;
+    let wallet = Wallet::builder()
+        .with_secret_manager(SecretManager::Mnemonic(secret_manager))
+        .with_storage_path(&var("WALLET_DB_PATH").unwrap())
+        .with_client_options(client_options)
+        .with_coin_type(SHIMMER_COIN_TYPE)
+        .finish()
+        .await?;
+
+    // Ensure there's some base coin balance
+    let account = get_or_create_account(&wallet, ACCOUNT_ALIAS).await?;
+    let addresses = generate_max_addresses(&account, NUM_CONCURRENT_TXS).await?;
+
+    // Ensure there are enough available funds for spamming on each address
+    let amount = may_request_funds(&account, addresses[0].address().to_string().as_str()).await?;
+    account.sync(None).await?;
+
+    let num_addresses_with_unspent_outputs = account.addresses_with_unspent_outputs().await?.len();
+    println!("Addresses with balance count (before): {num_addresses_with_unspent_outputs}",);
+
+    println!("Splitting available funds...");
+    let split_amount = amount / NUM_CONCURRENT_TXS as u64;
+
+    for address in addresses.iter().map(|addr| addr.address().to_string()) {
+        println!("Sending {split_amount} to {address}");
+        let outputs = vec![SendAmountParams::new(address, split_amount)];
+        let transaction = account.send_amount(outputs, None).await?;
+
+        println!(
+            "Transaction sent: {}/transaction/{}",
+            var("EXPLORER_URL").unwrap(),
+            transaction.transaction_id
+        );
+
+        // Wait for transaction to get included
+        let block_id = account
+            .retry_transaction_until_included(&transaction.transaction_id, None, None)
+            .await?;
+
+        println!(
+            "Transaction included: {}/block/{}",
+            var("EXPLORER_URL").unwrap(),
+            block_id
+        );
+
+        account.sync(None).await?;
+    }
```

Comment:
This should just be a single transaction with 127 (so we have 1 left for remainder) outputs to the first address

Reply:
ah, of course :+1:

Reply:
Done